src/Transformations.jl

```diff
@@ -260,6 +261,64 @@ function transform!(tn::TensorNetwork, config::ColumnReduction)
     return tn
 end
 
+Base.@kwdef struct SplitSimplification <: Transformation
+    atol::Float64 = 1e-10  # A threshold for SVD rank determination
+end
+
+function transform!(tn::TensorNetwork, config::SplitSimplification)
+    done = false  # This flag will be set to true when no further changes are made
+
+    while !done
+        done = true  # Assume no changes will be made this pass
+
+        # Create a copy of the list of tensors so we can modify the original while iterating
+        tensor_list = copy(tensors(tn))
+
+        for tensor in tensor_list
```

Comment:
I would prefer if you do something with `@label` and `@goto` like in `RankSimplification`.

Reply:
Okay, I will take a look into `RankSimplification`